gc_registry/certificate/services.py

```diff
@@ -0,0 +1,59 @@
+from hashlib import sha256
+
+from .models import (
+    GranularCertificateBundle,
+    GranularCertificateBundleBase,
+)
+from .schemas import GranularCertificateBundleCreate
+
+
+def validate_transfer():
+    pass
+
+
+def create_bundle_hash(
```

Comment:
@Salamehar any thoughts on this from your PhD?!

Reply:
If the nonce defaults to an empty string, it may lead to hash collisions or unintended lineage chains.

Make the nonce parameter required to enforce explicit lineage chaining.